### PR TITLE
feat: add form component

### PR DESCRIPTION
**Pull request by Spacelocust**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [X] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: front/app/lib/utils/zod.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { z } from 'zod';
+
+import type { ZodErrorMap } from 'zod';
+
+type ParamsZod = {
```
Comment (YummYume) [inline front/app/lib/utils/zod.ts]: Possible de faire un typeof à la place ?

File: front/app/lib/utils/dayjs.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import utc from 'dayjs/plugin/utc';
+
+// Plugins
+dayjs.extend(utc);
+dayjs.extend(timezone);
+dayjs.extend(isSameOrBefore);
+dayjs.extend(isSameOrAfter);
+
+// Defaults
+dayjs.tz.setDefault('Europe/Paris');
```
Comment (YummYume) [inline front/app/lib/utils/dayjs.ts]:
```suggestion
dayjs.tz.setDefault('America/New_York');
```

File: front/app/lib/utils/dayjs.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export default dayjs;
```
Comment (YummYume) [inline front/app/lib/utils/dayjs.ts]: On peut peut-être faire un `export { dayjs }` parce que sinon l'auto import a du mal

File: front/app/lib/api/user/index.ts [hunk 4]
```diff
@@ -17,12 +24,29 @@ export const getCurrentUser = async (client: GraphQLClient) => {
...
 
-export const requestAuthToken = async (
-  client: GraphQLClient,
-  username: string,
-  password: string,
-) => {
-  return client.request<Mutation>(
+type RegisterInput = Required<Omit<CreateUserInput, 'clientMutationId'>>;
+
+export const requestRegister = async (client: GraphQLClient, registerInput: RegisterInput) => {
```
Comment (YummYume) [inline front/app/lib/api/user/index.ts]:
```suggestion
export const createRegistrationDemand = async (client: GraphQLClient, registerInput: RegisterInput) => {
```

File: front/app/lib/components/form/custom/FieldInput.tsx [hunk 5]
```diff
@@ -13,20 +18,27 @@ export type FieldInputProps = {
 } & React.ComponentProps<typeof TextField.Input> &
   React.RefAttributes<HTMLInputElement>;
 
-export const FieldInput = ({
+export function FieldInput<T extends FormData>({
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldInput.tsx:24]: Tu peux laisser le const et mettre le type générique avant la ( je crois

File: front/app/routes/_landing/forgotten-password/route.tsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ export default function ForgottenPassword() {
       </Heading>
       <Container p="4" size="1">
         <Form method="post" className="mb-4 space-y-4" unstable_viewTransition>
-          <FieldInput name="username" label={t('username')} type="text" error="" required />
+          {/* <FieldInput name="username" label={t('username')} type="text" error="" required /> */}
```
Comment (YummYume) [inline front/app/routes/_landing/forgotten-password/route.tsx]: Oublie ?

File: front/app/lib/components/form/custom/FieldInput.tsx [hunk 7]
```diff
@@ -13,20 +18,27 @@ export type FieldInputProps = {
...
+}: FieldInputProps<T>) {
+  const { t } = useTranslation();
+  const {
+    register,
+    formState: { errors },
+  } = useRemixFormContext<T>();
+
   const ariaDescribedBy = `${name}-error`;
 
+  const error = (errors[name]?.message ? errors[name]?.message : null) as string | null;
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldInput.tsx]: Possible de faire une fonction plutôt ou quelque chose pour que ce soit plus clean ?

File: front/app/lib/components/form/custom/FieldInput.tsx [hunk 8]
```diff
@@ -36,17 +48,17 @@ export const FieldInput = ({
...
           aria-describedby={error ? ariaDescribedBy : ''}
+          {...register(name)}
           {...rest}
         />
         {rightSlot}
       </TextField.Root>
       {error && (
         <Text as="p" id={ariaDescribedBy} className={errorClassName}>
-          {error}
+          {t(error, { ns: 'validators' })}
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldInput.tsx:62]: Du coup si une erreur vient de l'API, il va quand même essayer de la traduire ?

File: front/app/lib/components/form/custom/FieldInput.tsx (hunk 5 above)
Comment (Spacelocust) [reply front/app/lib/components/form/custom/FieldInput.tsx:24]: Ouais mais je peux pas utiliser la générique dans le composant après

**Review by YummYume — APPROVED** [review]

**Pull request merged**